Automattic/wp-calypso
# A4A: add new referral calculations

Resolves https://github.com/Automattic/automattic-for-agencies-dev/issues/1581

## Proposed Changes

This PR adds new methods to calculate commissions from referrals. 
More details on the desired solution: pfunGA-3Wy-p2#comment-7052

We base our calculation only on active/canceled referral purchases. We calculate how much days it were (or will be) active during the activity window for the next payout date.

Additionally, tests are added to confirm calculations.

Note: it will be wired with UI later

## Why are these changes being made?



## Testing Instructions



- Checkout this branch locally
- Check the code
- Check test cases
- Run tests `yarn test-client client/a8c-for-agencies/sections/referrals/lib`

## Pre-merge Checklist
